a3/basic-scripting-guide.md

```diff
@@ -0,0 +1,406 @@
+---
+title: Lab 3 - Shell Scripting
+layout: lab
+---
+
+## Overview
+Many of the tasks that someone would like to perform on a computer are regular, require repetition, or are menial or tedious to do by hand. Shell scripting allows one to interact programmatically with a shell to do certain tasks. For example, the command for scanning log files in the previous topic guide could be automated to be performed on a schedule by means of a shell script. `bash` scripts are an incredibly powerful tool for sysadmins to automate tasks that are otherwise
+difficult to remember or long-running.
+
+In cases where shell syntax is inappropriate for the task at hand, one can instead call into programs written in other languages, such as Python, which can read from stdin, process data, and write to stdout. 
+
+---
+
+## What's Covered
+* `bash` as a scripting language
+* `python` for system administration
+
+## Scripting with the Bourne-Again Shell (Bash)
+While most programmers are likely familiar with `bash` in its popular capacity as a command line interpreter, it is in fact a powerful and full-featured programming language. Moreover, `bash` has a uniquely qualified claim to the title of *scripting* language in that programs written in `bash` are simply series of shell commands which `bash` reads off and executes line-by-line. Or, conversely, one might say that `bash` command line entries are simply short one-line scripts. So really, you’ve been `bash` scripting all along!
+
+### Shebang!
+Shell scripts typically begin with the **shebang** line: `#!pathtointerpreter`.
+
+[1]: https://en.wikipedia.org/wiki/Shebang_(Unix)#Magic_number
+`#!` is a human-readable representation of a [magic number][1] `0x23 0x21` used by the operating system to determine whether a file is a script or
```

Comment:
wait, doesn't the shell do this interpretation, not the OS? The shell interprets the command and then execs the appropriate binary to interpret the rest of the script